compiler/src/iree/compiler/Codegen/Common/LinalgOpInfo.cpp

```diff
@@ -99,9 +134,40 @@ static bool computeReductionInfo(LinalgOp linalgOp) {
   return linalgOp.getNumReductionLoops() > 1;
 }
 
+static SmallVector<OpOperand *> computeSharedMemTransposeInfo(
+    LinalgOp LinalgOp) {
+  SmallVector<OpOperand *> sharedMemTransposeOperands;
+
+  if (!isa<linalg::GenericOp>(LinalgOp)) {
+    return sharedMemTransposeOperands;
+  }
+
+  if (!checkTransposePreconditions(LinalgOp)) {
+    return sharedMemTransposeOperands;
+  }
+
+  // To simplify logic, we only consider linalg ops with transposes who's
+  // outputs maps are identities
+  for (OpOperand *opOperand : LinalgOp.getOutputOperands()) {
+    if (!LinalgOp.getTiedIndexingMap(opOperand).isIdentity()) {
+      return sharedMemTransposeOperands;
+    }
+  }
+
```

Comment:
Same here. I think all these checks above should be part of the LinalgOp lambda function.